### PR TITLE
wrap errors in selectorspread and podtoplogyspread plugin

### PR DESCRIPTION
**Pull request by arghya88**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Wrap errors in selectorspread and podtoplogyspread plugin

**Which issue(s) this PR fixes**:

Ref #94695

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go [hunk 1]
```diff
@@ -1651,7 +1652,7 @@ func TestPreFilterDisabled(t *testing.T) {
 	p := &PodTopologySpread{}
 	cycleState := framework.NewCycleState()
 	gotStatus := p.Filter(context.Background(), cycleState, pod, nodeInfo)
-	wantStatus := framework.NewStatus(framework.Error, `error reading "PreFilterPodTopologySpread" from cycleState: not found`)
+	wantStatus := framework.AsStatus(fmt.Errorf(`error reading "PreFilterPodTopologySpread" from cycleState: %w`, fmt.Errorf("not found")))
```
Comment (chendave) [inline pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go:1655]: what's the difference with this? anything more bring us?
```golang
framework.AsStatus(fmt.Errorf(`error reading "PreFilterPodTopologySpread" from cycleState: not found`))
```

Comment (arghya88) [reply pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go:1655]: I was following it up from another PR..but I agree and will revert this back

File: pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go [hunk 2]
```diff
@@ -1651,7 +1652,7 @@ func TestPreFilterDisabled(t *testing.T) {
 	p := &PodTopologySpread{}
 	cycleState := framework.NewCycleState()
 	gotStatus := p.Filter(context.Background(), cycleState, pod, nodeInfo)
-	wantStatus := framework.NewStatus(framework.Error, `error reading "PreFilterPodTopologySpread" from cycleState: not found`)
+	wantStatus := framework.AsStatus(fmt.Errorf(`error reading "PreFilterPodTopologySpread" from cycleState: not found`))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go:1655]: remove the word "error" from the message. Same for the rest of the PR.

File: pkg/scheduler/framework/plugins/podtopologyspread/scoring.go [hunk 3]
```diff
@@ -174,13 +174,13 @@ func (pl *PodTopologySpread) PreScore(
 func (pl *PodTopologySpread) Score(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
 	nodeInfo, err := pl.sharedLister.NodeInfos().Get(nodeName)
 	if err != nil || nodeInfo.Node() == nil {
-		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("getting node %q from Snapshot: %v, node is nil: %v", nodeName, err, nodeInfo.Node() == nil))
+		return 0, framework.AsStatus(fmt.Errorf("getting node %q from Snapshot: %w, node is nil: %v", nodeName, err, nodeInfo.Node() == nil))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/podtopologyspread/scoring.go:177]: `%w` should be at the end of the message.

IIRC, `nodeInfo.NodeI()` can no longer be nil, as we do the check when we create the Snapshot. Would you mind verifying and removing that check?

File: pkg/scheduler/framework/plugins/selectorspread/selector_spread.go [hunk 4]
```diff
@@ -86,17 +86,17 @@ func (pl *SelectorSpread) Score(ctx context.Context, state *framework.CycleState
...
 	c, err := state.Read(preScoreStateKey)
 	if err != nil {
-		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("Error reading %q from cycleState: %v", preScoreStateKey, err))
+		return 0, framework.AsStatus(fmt.Errorf("Error reading %q from cycleState: %w", preScoreStateKey, err))
 	}
 
 	s, ok := c.(*preScoreState)
 	if !ok {
-		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("%+v convert to tainttoleration.preScoreState error", c))
+		return 0, framework.AsStatus(fmt.Errorf("%+v convert to tainttoleration.preScoreState error", c))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/selectorspread/selector_spread.go:94]: we shouldn't reach this line, but "%+v" is very unreadable. Just remove it.

"cannot convert saved state to tainttoleration.preScoreState"

File: pkg/scheduler/framework/plugins/podtopologyspread/scoring.go (hunk 3 above)
Comment (arghya88) [reply pkg/scheduler/framework/plugins/podtopologyspread/scoring.go]: There is a check here
 https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/podtopologyspread/scoring.go#L142

File: pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go (hunk 2 above)
Comment (arghya88) [reply pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go]: Done

File: pkg/scheduler/framework/plugins/selectorspread/selector_spread.go (hunk 4 above)
Comment (arghya88) [reply pkg/scheduler/framework/plugins/selectorspread/selector_spread.go]: Done

File: pkg/scheduler/framework/plugins/podtopologyspread/scoring.go (hunk 3 above)
Comment (arghya88) [reply pkg/scheduler/framework/plugins/podtopologyspread/scoring.go]: Done

File: pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go (hunk 1 above)
Comment (arghya88) [reply pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go]: Done

File: pkg/scheduler/framework/plugins/podtopologyspread/scoring.go (hunk 3 above)
Comment (alculquicondor) [reply pkg/scheduler/framework/plugins/podtopologyspread/scoring.go]: /hold

The source of truth is actually the scheduler cache https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/internal/cache/cache.go#L201

And it looks like there might actually be a bug there were we consider some Nodes that are nil. I'm evaluating.

**Comment by arghya88** [comment]
remove do-not-merge/hold

**Comment by alculquicondor** [comment]
/hold

Hold was intended

**Comment by alculquicondor** [comment]
/hold cancel
because #95130 is merged

**Pull request merged**